docs/certificate-specification.md

```diff
@@ -0,0 +1,125 @@
+# Certificate Specification
+
+This document includes the requirements for root, intermediate, and issued certificates.
+This document applies to all instances of Fulcio, including the production instance and
+all private instances using the service defined in this repository.
+
+The key words "MUST", "MUST NOT", "SHOULD", "SHOULD NOT", and "MAY" in this document are
+to be interpreted as described in [RFC 2119](https://www.ietf.org/rfc/rfc2119.txt).
+
+## Root Certificate
+
+A root certificate MUST:
+
+* Specify a Subject with a common name and organization
+* Specify an Issuer with the same values as the Subject
+* Specify Key Usages for Certificate Sign and CRL Sign
+* Specify Basic Constraints to `CA:TRUE`
+* Specify a unique, positive, 160 bit serial number
```

Comment:
Maybe too nitpicky, but can we offer some normative `MAY` guidance for generating this serial?

Reply:
Do you mean how to generate it, as in what's the best way to generate a random 160 bit number, or do you mean to relax this requirement? For the latter, this is coming from RFC5280, I just called it out specifically since it's a less well-known part of 5280.

Reply:
The former 🙂 -- I wanted to understand if we're asking for the number to be *random* (~approximating no numeric relationship between certificates) or merely *unique* (never repeated, but perhaps a simple increasing integer).

If it's meant to be like 5280 (https://datatracker.ietf.org/doc/html/rfc5280#section-4.1.2.2), then maybe qualify with "according to the same rules as RFC 5280 4.1.2.2"?

Reply:
Random is recommended, IIRC there have been some issues in the past with monotonically increasing values. I'll note this, and add the qualification.